cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -159,13 +168,16 @@ void DrawableSubmap::SetAlpha(const double current_tracking_z) {
 
 void DrawableSubmap::UpdateSceneNode() {
   ::cartographer::common::MutexLocker locker(&mutex_);
+  render_in_progress_ = false;
+  if (quitting_) {
```

Comment:
I started preparing the updated version of this pull request for you, and just before pushing everything as we agreed on above, I remembered another reason why I introduced the splitting of `query_in_progress_`. 

In the second pull request, where you can toggle individual submap visibility, I added deferring the texture update in `UpdateSceneNode()` for hidden submaps. Only when you toggle their visibility back, the additional call of `UpdateSceneNode()`, which results from clicking the checkbox (and not from a finished fetch thread) will update their texture. This has been added in commit 6f4095b226e1.

I now find setting `query_in_progress_ = false` in `UpdateSceneNode()` a bit awkward, as there could be another query in progress, unrelated to user's clicking of the checkbox. Damon wanted these two pull requests separated, so the splitting made less sense on its own in this pull request.

With that in mind, the options are now the following:
 *  discard commit 6f4095b226e1 in the other pull request, and always fully run `UpdateSceneNode()`, no matter whether the submap is hidden; this pull request is updated as we agreed above
  * we keep the two separate flags, while removing `quitting_`, and keep the deferred texture update in the other pull request